J1032 vsm: do not abort when signal is None

gctucker opened the pull request:
When an IPC module receives a signal it doesn't know how to decode, it
should ignore it and its receive() method should return None.  When
this happens, skip the unknown signal in the VSM main loop.  If it
should have been decoded and passed to the VSM but failed, the
relevant IPC module can decide to raise an exception to actually
abort.

Signed-off-by: Guillaume Tucker <guillaume.tucker@collabora.com>

gctucker committed:
vsm: do not abort when signal is None

When an IPC module receives a signal it doesn't know how to decode, it
should ignore it and its receive() method should return None.  When
this happens, skip the unknown signal in the VSM main loop.  If it
should have been decoded and passed to the VSM but failed, the
relevant IPC module can decide to raise an exception to actually
abort.

Signed-off-by: Guillaume Tucker <guillaume.tucker@collabora.com>

kov reviewed (approved):
Was this behaviour being tested? May need to update tests. What about logging? This sounds like the kind of thing we may want to log at debugging level, for when we try to figure out issues?

gctucker commented:
OK I can add a unit test with an invalid message.  I don't think this was tested before.

It was not logged before, so if the IPC module returned None the VSM would exit without any indication of what happened.  So yes it could be logged, but it could already have been logged before :)  I'll add that as well.

gctucker committed:
tests: add VSMNoneSignalTests

Add a new test suite VSMNoneSignalTests with a modified version of
StdioIPC to test the case where the IPC module returns a None rather
than a (signal, value) pair.

Signed-off-by: Guillaume Tucker <guillaume.tucker@collabora.com>

gctucker commented:
Note: This should be merged after #47.
